lib/std/priority_queue.zig

```diff
@@ -18,33 +18,26 @@ pub fn PriorityQueue(comptime T: type, comptime Context: type, comptime compareF
     return struct {
         const Self = @This();
 
-        items: []T,
-        cap: usize,
-        allocator: Allocator,
+        items: []T = undefined,
+        /// The capacity of the queue. This may be read directly, but must not
+        /// be modified directly.
+        capacity: usize = 0,
         context: Context,
 
-        /// Initialize and return a priority queue.
-        pub fn init(allocator: Allocator, context: Context) Self {
```

Comment:
i think that this should still use an init function rather than default values, if you were to for example initialize it with `.{ .capacity = 5, .context = .{} }` this would be invalid as the `items` field wouldnt correctly fit the given capacity. see <https://ziglang.org/documentation/master/#Faulty-Default-Field-Values>

Reply:
this rule/guideline/whatever you might call it regarding default values is fairly new, im not sure why multiarraylist is still using default values but i think it really shouldnt, a few other types have started moving in the same direction https://github.com/ziglang/zig/commit/0b9fccf508dc85fa522947d1cf6ff84f78f2dcb4